app/workout/[id].tsx

```diff
@@ -0,0 +1,30 @@
+import { useLocalSearchParams } from 'expo-router';
+import { Text, View, StyleSheet } from 'react-native';
+import React from 'react';
+import { getWorkout } from '@/hooks/db';
+import { QueryClient, QueryClientProvider, useMutation, useQuery } from '@tanstack/react-query';
+import { WorkoutType } from '@/hooks/types';
+import { workoutsQueryKey } from '@/hooks/query';
+import { Workout } from '@/components/Workout';
+
+export const queryClient = new QueryClient();
+
+export default function Page() {
+  const { id } = useLocalSearchParams();
+
+  return (
+    <QueryClientProvider client={queryClient}>
```

Comment:
I dont think you want another QueryClientProvider. You want just one in your app. At the top level